### DISCUSSION TITLE
关于代理设置的一个问题（A problem about setting proxy）

### DISCUSSION THREAD

**Discussion by BUAAYurika** [binary-husky/gpt_academic#310] [Q&A]
感谢各位开发者提供了这样一款优秀的软件。

我计划在中国内地使用它，所以我需要设置代理。我使用的是v2ray软件，而且可以正常访问谷歌学术以及chatgpt网页。按照说明，我在v2ray软件的设置中查到端口号为10810。于是，我将config.py文件中的proxy设置为：
`proxies = { 
        #          [协议]://  [地址]  :[端口]
        "http":  "socks5h://127.0.0.1:10810", 
        "https": "socks5h://127.0.0.1:10810", 
    }`
然而，我使用check_proxy.py测试仍然报告“代理配置 socks5h://127.0.0.1:10810, 代理所在地查询超时，代理可能无效”。我也尝试了打开Internet设置的方法，发现其中的端口号为10811。我接着尝试了将协议改为http，以及分别与两个端口号各自的组合，但这些都无法连通。请问我还要改修改哪些设置？谢谢！

Thank you developers for providing such an excellent software.

I plan to use it in mainland China, so I need to set up a proxy. I am using the v2ray software, and I can access Google Scholar and ChatGPT web pages normally. According to the instructions, I found the port number to be 10810 in the v2ray software settings. So I set the proxy in the config.py file as follows:

`proxies = { 
        #          [协议]://  [地址]  :[端口]
        "http":  "socks5h://127.0.0.1:10810", 
        "https": "socks5h://127.0.0.1:10810", 
    }`

However, when I tested it by check_proxy.py, it still reported "Proxy configuration socks5h://127.0.0.1:10810, proxy location query timed out, the proxy may be invalid". I also tried the method of opening Internet settings and found that the port number in it was 10811. I then tried changing the protocol to http and combining each with the two port numbers respectively, but none of these worked. Could you please advise me on what other settings I need to modify? Thank you!

（Translated by ChatGPT）

**Comment by binary-husky** [comment]
https://github.com/binary-husky/chatgpt_academic/wiki/%E4%BB%A3%E7%90%86%E8%BD%AF%E4%BB%B6%E9%97%AE%E9%A2%98%E7%9A%84%E6%96%B0%E6%89%8B%E8%A7%A3%E5%86%B3%E6%96%B9%E6%B3%95%EF%BC%88%E6%96%B9%E6%B3%95%E5%8F%AA%E9%80%82%E7%94%A8%E4%BA%8E%E6%96%B0%E6%89%8B%EF%BC%89

**Reply by zhengzhouMU** [reply]
我也遇到了同样的问题。不过我按照纯新手教程解决了。

**Reply by vampireseraph** [reply]
我是软路由passwall科学的，使用127.0.0.1:1070（TCP 节点 Socks 监听端口）无法成功代理，请问该如何解决呢？

> https://github.com/binary-husky/chatgpt_academic/wiki/%E4%BB%A3%E7%90%86%E8%BD%AF%E4%BB%B6%E9%97%AE%E9%A2%98%E7%9A%84%E6%96%B0%E6%89%8B%E8%A7%A3%E5%86%B3%E6%96%B9%E6%B3%95%EF%BC%88%E6%96%B9%E6%B3%95%E5%8F%AA%E9%80%82%E7%94%A8%E4%BA%8E%E6%96%B0%E6%89%8B%EF%BC%89

**Reply by Jrunchang** [reply]
谢谢老哥

**Reply by Jrunchang** [reply]
请问在LAN设置中，对于本地地址不使用代理服务器，这个关不掉（√），运行main.py出问题

**Comment by zhengzhouMU** [comment]
你好
&nbsp; &nbsp; &nbsp;请问问题解决了没


刘帅峰
***@***.***

安徽大学


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "binary-husky/chatgpt_academic"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月16日(星期天) 上午10:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [binary-husky/chatgpt_academic] 关于代理设置的一个问题（A problem about setting proxy） (Discussion #310)





 
我是软路由passwall科学的，使用127.0.0.1:1070（TCP 节点 Socks 监听端口）无法成功代理，请问该如何解决呢？
  
https://github.com/binary-husky/chatgpt_academic/wiki/%E4%BB%A3%E7%90%86%E8%BD%AF%E4%BB%B6%E9%97%AE%E9%A2%98%E7%9A%84%E6%96%B0%E6%89%8B%E8%A7%A3%E5%86%B3%E6%96%B9%E6%B3%95%EF%BC%88%E6%96%B9%E6%B3%95%E5%8F%AA%E9%80%82%E7%94%A8%E4%BA%8E%E6%96%B0%E6%89%8B%EF%BC%89

**Comment by zhengzhouMU** [comment]
你好
&nbsp; &nbsp; &nbsp;运行main.py出问题。跟这个”对于本地地址不使用代理服务器，这个关不掉（√）“没有关系。你把 pyhon main.py 报错的原因看一下




刘帅峰
***@***.***

安徽大学


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "binary-husky/chatgpt_academic"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月23日(星期天) 晚上8:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [binary-husky/chatgpt_academic] 关于代理设置的一个问题（A problem about setting proxy） (Discussion #310)





 
请问在LAN设置中，对于本地地址不使用代理服务器，这个关不掉（√），运行main.py出问题

**Comment by Jrunchang** [comment]
谢谢您的回复，已经解决了。之前我尝试远程ssh 连接linux服务器，在上面配置。都是代理所在地查询超时，代理可能无效”，这个错误。应该是服务器上的proxy和本地的不同，后来在本地D盘配置，10分钟就成功了，多谢，给您star了。

**Comment by zhengzhouMU** [comment]
你好
&nbsp; &nbsp; &nbsp;我觉得这个chatGPT，没有想象中的那么强大。不知道你怎么看，还是我运用的方式不对




刘帅峰
***@***.***

安徽大学


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "binary-husky/chatgpt_academic"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月24日(星期一) 中午11:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [binary-husky/chatgpt_academic] 关于代理设置的一个问题（A problem about setting proxy） (Discussion #310)





 
谢谢您的回复，已经解决了。之前我尝试远程ssh 连接linux服务器，在上面配置。都是代理所在地查询超时，代理可能无效”，这个错误。应该是服务器上的proxy和本地的不同，后来在本地D盘配置，10分钟就成功了，多谢，给您star了。

**Reply by Jrunchang** [reply]
还行，对平时阅读文献 和code有帮助
